tests/DocBlock/TypeExpressionTest.php

```diff
@@ -234,6 +234,17 @@ public static function provideGetTypesCases(): iterable
         yield ['array{a: int, b: int, c: int, d: int, e: int, f: int, g: int, h: int, i: int, j: int, with-dash: int}'];
 
         yield ['array{a: int, b: int, c: int, d: int, e: int, f: int, g: int, h: int, i: int, j: int, k: int, l: int, with-dash: int}'];
+
+        yield [sprintf(
+            'array{%s}',
+            implode(
+                ', ',
+                array_map(
+                    static fn (int $k): string => sprintf('k%s: int', $k),
```

Comment:
Maybe we can involve the dash here too? Not necessarily to all keys, maybe only to odd/even?

Reply:
Then that test will fail 😄. Maybe it is time to consider using https://github.com/phpstan/phpdoc-parser.